lib/ash_phoenix/form/form.ex

```diff
@@ -394,6 +394,60 @@ defmodule AshPhoenix.Form do
 
   import AshPhoenix.FormData.Helpers
 
+  @behaviour Access
+
+  @doc false
+  @impl Access
+  def fetch(%__MODULE__{} = _form, key) do
+    raise """
+    Cannot access AshPhoenix.Form.
+
+    You're trying to access a form field using: form[#{inspect(key)}]
+
+    If you are using Phoenix, you are likely missing to_form/2 call on the form.
+
+    Instead of:
+        AshPhoenix.Form.for_create(MyApp.Blog.Post, :create)
+
+    Use:
+        AshPhoenix.Form.for_create(MyApp.Blog.Post, :create) |> to_form()
+    """
+  end
+
+  @impl Access
+  def get_and_update(%__MODULE__{} = _form, key, _function) do
+    raise """
+    Cannot access AshPhoenix.Form.
+
+    You're trying to access a form field using: form[#{inspect(key)}]
+
+    If you are using Phoenix, you are likely missing to_form/2 call on the form.
+
+    Instead of:
+        AshPhoenix.Form.for_create(MyApp.Blog.Post, :create)
+
+    Use:
+        AshPhoenix.Form.for_create(MyApp.Blog.Post, :create) |> to_form()
+    """
+  end
+
+  @impl Access
+  def pop(%__MODULE__{} = _form, key) do
+    raise """
+    Cannot access AshPhoenix.Form.
```

Comment:
Let's store this in a module attribute or in a content somewhere and refer to that so we don't have the message repeated.